src/pepr/operator/crd/validators/package-validator.ts

```diff
@@ -141,5 +141,24 @@ export async function validator(req: PeprValidateRequest<UDSPackage>) {
     }
   }
 
+  const monitors = pkg.spec?.monitor ?? [];
+
+  // Ensure serviceMonitors use a unique description or selector/portName used for generating the resource name
+  const monitorNames = new Set<string>();
+
+  for (const monitor of monitors) {
+    const monitorName = sanitizeResourceName(
+      monitor.description || `${Object.values(monitor.selector)}-${monitor.portName}`,
+    );
+
+    if (monitorNames.has(monitorName)) {
+      return req.Deny(
+        `A serviceMonitor resource generated from the Package, ${pkg.metadata?.name} contains a duplicate name, please provide a unique description for each item in the monitor array`,
+      );
```

Comment:
nit: might be good to align this with the wording we use for the [VirtualService check](https://github.com/defenseunicorns/uds-core/blob/monitor-validate/src/pepr/operator/crd/validators/package-validator.ts#L44-L48), and make sure we are capturing `kind` accurately here (pod vs svc monitor).